realm/realm-library/src/main/cpp/CMakeLists.txt

```diff
@@ -131,32 +130,19 @@ create_javah(TARGET jni_headers
     DEPENDS ${classes_PATH}
 )
 
-include(RealmCore)
-use_realm_core(${build_SYNC} "${REALM_CORE_DIST_DIR}" "${CORE_SOURCE_PATH}")
-
-# Download OpenSSL lib
-# FIXME Read the openssl version from core when the core/sync release has that information.
-set(openssl_VERSION "1.1.1b")
-set(openssl_FILENAME "openssl.tgz")
-set(openssl_URL "https://static.realm.io/downloads/openssl/${openssl_VERSION}/Android/${ANDROID_ABI}/${openssl_FILENAME}")
-
-message(STATUS "Downloading OpenSSL...")
-file(DOWNLOAD "${openssl_URL}" "${PROJECT_BINARY_DIR}/${openssl_FILENAME}")
-
-message(STATUS "Uncompressing OpenSSL: ${PROJECT_BINARY_DIR}/${openssl_FILENAME}")
-execute_process(COMMAND ${CMAKE_COMMAND} -E tar xfz "${openssl_FILENAME}" WORKING_DIRECTORY "${PROJECT_BINARY_DIR}")
-message(STATUS "Importing OpenSSL...")
-include(${PROJECT_BINARY_DIR}/lib/cmake/OpenSSL/OpenSSLConfig.cmake)
-get_target_property(openssl_include_DIR OpenSSL::Crypto INTERFACE_INCLUDE_DIRECTORIES)
-get_target_property(crypto_LIB OpenSSL::Crypto IMPORTED_LOCATION)
-get_target_property(ssl_LIB OpenSSL::SSL IMPORTED_LOCATION)
+option(REALM_JAVA_BUILD_CORE_FROM_SOURCE "Build Realm Core from source, as opposed to downloading prebuilt binaries" ON)
+if(REALM_JAVA_BUILD_CORE_FROM_SOURCE)
+    set(REALM_BUILD_LIB_ONLY ON)
+    add_subdirectory(realm-core)
+else()
+    message(FATAL_ERROR "TODO: Implement downloading monorepo release artifacts")
```

Comment:
@fealebenpae Any idea about a timeline when these are available?

Reply:
I want to iron out all the kinks in the integration before I cut binary releases, so as soon as I've fixed the other issues (like the Mixed serialization above).